tripleblindmarket/covid-safe-paths
# Refactor settings styles

## Description

This consolidates settings styles to use `<Theme>`, `<Typography>` and `<Typography secondary>` to closer match theme styles from Figma.

* **Remove the duplicated Google import block!**
* Wraps the entire app in a `<Theme use="default">` so that any `<Typography>` below is correctly colored.
* Extract out some common logic for language detection and picker.
* Extracts out common `<Section>` and `<SettingsItem>` components, reduce file size, keep styles and components consistent, small and encapsulated.
* Converts to using hooks because those are the new normal for React.
* Uses styled components generally.They are theme aware, and are better tracked via eslint:
  * A missing style is not shown in eslint, but is clear with styled components.
  * Unused styled components are captured by eslint, though this is also captured with StyleSheet.create()

## How to test

Check for any visual artifacts on settings page.

Fonts are changed, this is intentional: Settings header is not bold, settings description text is secondary and smaller.

Google import text below the button should be monospace and secondary color

Before:
<img width="361" alt="Screen Shot 2020-04-17 at 1 58 25 PM" src="https://user-images.githubusercontent.com/702990/79613608-b86eec80-80b3-11ea-9d1a-fd1b4e5ff7c7.png">

After:
<img width="309" alt="Screen Shot 2020-04-17 at 1 57 43 PM" src="https://user-images.githubusercontent.com/702990/79613655-c91f6280-80b3-11ea-8002-6f46818adfec.png">
